gateway-ha/src/test/java/io/trino/gateway/CustomImageNameSubstitutor.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.gateway;
+
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.ImageNameSubstitutor;
+
+public class CustomImageNameSubstitutor
+        extends ImageNameSubstitutor
+{
+    private static final String TESTCONTAINERS_TRINO_IMAGE_SUBSTITUTE = "TESTCONTAINERS_TRINO_IMAGE_SUBSTITUTE";
+
+    @Override
+    public DockerImageName apply(DockerImageName dockerImageName)
+    {
+        if (dockerImageName.asCanonicalNameString().equals("trinodb/trino:latest") && System.getenv(TESTCONTAINERS_TRINO_IMAGE_SUBSTITUTE) != null) {
```

Comment:
Remove the 1st condition.

Reply:
first condition makes sure we are overriding a specific image. If we don't specify it will just substitute all images

Reply:
See https://github.com/trinodb/trino-gateway/pull/266#discussion_r1505078523

Reply:
I understand you want to change only Trino image names. I suppose we should tweak the condition to exclude the version.

Reply:
we can probably use this `getUnversionedPart()` from https://javadoc.io/doc/org.testcontainers/testcontainers/1.15.1/org/testcontainers/utility/DockerImageName.html#getUnversionedPart--